eth/eventhandler/handlers.go

```
@@ -436,6 +450,13 @@ func (eh *EventHandler) handleClusterReactivated(txn basedb.Txn, event *contract
 			return nil, fmt.Errorf("could not bump slashing protection: %w", err)
 		}
 
+		// Apply any archived slashing protection data for this validator.
+		// TODO(SSV-15): This ensures slashing protection continuity across validator reactivation cycles
+		// as part of the temporary solution for audit finding SSV-15.
+		if err := eh.keyManager.ApplyArchivedSlashingProtection(share.ValidatorPubKey[:], phase0.BLSPubKey(share.SharePubKey)); err != nil {
+			eh.logger.Warn("could not apply archived slashing protection data during reactivation", zap.Error(err))
+		}
```

I think we need to restore archived values **before** we call `eh.keyManager.BumpSlashingProtection` (otherwise I'm not sure what restoring archived values will do for us) ?

Also, I think we should probably return error instead of logging it as Warn - since we don't want to risk slashing (why have this protection otherwise if it's "best effort only")